django/researchdata/views.py

```diff
@@ -17,10 +17,17 @@ class StoryCreateSuccessTemplateView(TemplateView):
     """
     Class-based view to show the story create success template
     """
-
     template_name = 'researchdata/story-create-success.html'
 
 
+class StoryDetailView(DetailView):
+    """
+    Class-based view to show the story detail template
+    """
+    template_name = 'researchdata/story-detail.html'
+    queryset = models.Story.objects.filter(admin_published=True)
+
+
 class StoryListView(ListView):
     """
     Class-based view to show the story list template
```

Comment:
There should be a filter for `admin_published=True`. (I cannot add the comment to the relevant line below as it is not part of this PR.) I'm happy for that to be done in a future PR.

Reply:
This is an odd one, as the StoryListView actually loops through Locations (which don't have admin_published, as they're only managed by admins) and then in the template I find the stories for all locations. But when looping in the template, I've added a "if story.admin_published=True", so it'll still filter out stories that aren't approved.